sky/backends/onprem_utils.py

```diff
@@ -491,3 +492,41 @@ def save_distributable_yaml(cluster_config: Dict[str, Dict[str, Any]]) -> None:
     abs_yaml_path = os.path.expanduser(yaml_path)
     os.makedirs(os.path.dirname(abs_yaml_path), exist_ok=True)
     common_utils.dump_yaml(abs_yaml_path, cluster_config)
+
+
+# Currently, programmatic API doesn't check this.
+def check_local_cloud_args(cloud: Optional[str] = None,
```

Comment:
With #978 merged, should we move this to `sky.execution._execute`? That way we will also be able to perform these checks for the python API too. Please push back if it doesn't make sense

Note that this might require changing click.UsageErrors to standard python errors (ValueError?)

Reply:
It might not be possible! `check_local_cloud_args` is needed to be called before the `Task` object is created.

Reply:
Ah that makes sense. This is only for parsing YAMLs correctly then I believe.